### PR TITLE
Add St. Jude's HPCF Cluster Config

### PR DESCRIPTION
**Pull request by HaidYi**
---
name: StJude Config
about: St. Jude's HPCF Cluster Config
---

Please follow these steps before submitting your PR:

- [ ] If your PR is a work in progress, include `[WIP]` in its title
- [x] Your PR targets the `master` branch
- [x] You've included links to relevant issues, if any

Steps for adding a new config profile:

- [x] Add your custom config file to the `conf/` directory
- [x] Add your documentation file to the `docs/` directory
- [x] Add your custom profile to the `nfcore_custom.config` file in the top-level directory
- [x] Add your custom profile to the `README.md` file in the top-level directory
- [x] Add your profile name to the `profile:` scope in `.github/workflows/main.yml`

### CODE REVIEW

File: conf/stjude.config [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    config_profile_description = "St. Jude Children's Research Hospital HPC cluster (HPCF) profile"
+    config_profile_url = "https://www.stjude.org/"
+
+    max_cpus   = 32
+    max_memory = 1024.GB
+    max_time   = 240.h
+}
+
+process {
+    executor = 'lsf'
```
[the remark this answers is not in the document]
Comment (jfy133) [reply conf/stjude.config:21]: This is  a new thing as teh `max_` parameters are about to be deprecated in newer nf-core pipelines :)

[the remark this answers is not in the document]
Comment (jfy133) [reply conf/stjude.config:21]: (but keep those parameters for older pipelines)

File: docs/stjude.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+3. (Optional) If you want to use docker (unsupported) or singularity (supported) on HPCF, you need to add the following container options:
+
+```bash
+containerOptions = {
+    workflow.containerEngine == "singularity" ? '--nv':
+        ( workflow.containerEngine == "docker" ? '--gpus all': null )
+}
+```
+
+## Singularity Configuration
```
[the remark this answers is not in the document]
Comment (HaidYi) [reply docs/stjude.md:61]: Hi @jfy133, I have committed the suggestions you recommended and added the comments to them as well. For GPU section, I delete the global `param.num_gpus` parameter, set the number of gpu used in the `clusterOptions` based on the process-wise `task.accelerator`.

Could you review it again? Let me know if there are any other issues with it.

File: conf/stjude.config [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    // To use GPU for your nextflow process:
+    // 1. GPU processes need to set the `accelerator` option in their directives. This can be configured by:
+    //   1. Adding directives in each process specifically
+    //   2. Using `process {}`-based label selectors (e.g. label 'process_gpu')
+    //
+    // 2. (optional) If you want to use docker (unsupported) or singularity (supported) on HPCF, you need to add the following container options:
+    //   containerOptions = {
+    //       workflow.containerEngine == "singularity" ? '--nv':
+    //           ( workflow.containerEngine == "docker" ? '--gpus all': null )
+    //   }
```
Comment (maxulysse) [inline conf/stjude.config:70]: For me this should be more in the README for stjude than as comment in the config file

File: conf/stjude.config [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    autoMounts = true
+    runOptions = '-p -B "$TMPDIR"'
+    pullTimeout = "3 hours"
+
+    // Directory to cache singularity images. Unavailable with an institution-level set-up.
+    // FIXME: This needs the support from the cluster administrator.
+    // Feel free to use your own directory by:
+    //   1. use `singularity.cacheDir = /path/to/your/singularity/cache` in your config.
+    //   2. set env variable: `export NXF_SINGULARITY_CACHE=/path/to/your/singularity/cache` in your shell's profile file (e.g. `~/<.bashrc/.zshrc>`).
+}
```
Comment (maxulysse) [inline conf/stjude.config:87]: Same here

File: conf/stjude.config (hunk 3 above)
Comment (HaidYi) [reply conf/stjude.config]: Hi @maxulysse, thank you for the comments. I have moved the long comments from the config file to readme: `stjude.md`.

File: docs/stjude.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+- Environment variables whitelist: SINGULARITY_TMPDIR, TMPDIR, CUDA_VISIBLE_DEVICES
+- Pull timeout: 3 hours
+
+### Singularity Cache Directory
+
+There is no institution-level directory setup for caching singularity images. To set the cache directory, you
+can try the following two ways:
+
+1. Add `singularity.cacheDir = /path/to/your/singularity/cache` in your nextflow config file.
+2. Set nextflow's environment variable: `export NXF_SINGULARITY_CACHE=/path/to/your/singularity/cache` in your shell's profile file (e.g. `~/<.bashrc/.zshrc>`).
```
Comment (jfy133) [inline docs/stjude.md:80]: I would flip these personally, the env variable will be less likely over-written whereas NExtflow has some config order priority issues when you have too many

**Review by jfy133 — APPROVED** [review]
One list comment buto therwise LGTM now

**Pull request merged**